### ISSUE TITLE
💅 noFloatingPromises detection does not work with ESM modules

### ISSUE THREAD

**Issue by Zer0PointModule** [biomejs/biome#7600]
### Environment information

```bash
Using ESM modules and importing from .js files in a Typescript project noFloatingPromises does not detect floating promises.
```

### Rule name

noFloatingPromises

### Playground link

https://biomejs.dev/playground/?files.biome.json=ewAKAAkAIgAkAHMAYwBoAGUAbQBhACIAOgAgACIAaAB0AHQAcABzADoALwAvAGIAaQBvAG0AZQBqAHMALgBkAGUAdgAvAHMAYwBoAGUAbQBhAHMALwAyAC4AMgAuADQALwBzAGMAaABlAG0AYQAuAGoAcwBvAG4AIgAsAAoACQAiAHYAYwBzACIAOgAgAHsACgAJAAkAIgBlAG4AYQBiAGwAZQBkACIAOgAgAGYAYQBsAHMAZQAsAAoACQAJACIAYwBsAGkAZQBuAHQASwBpAG4AZAAiADoAIAAiAGcAaQB0ACIALAAKAAkACQAiAHUAcwBlAEkAZwBuAG8AcgBlAEYAaQBsAGUAIgA6ACAAZgBhAGwAcwBlAAoACQB9ACwACgAJACIAZgBpAGwAZQBzACIAOgAgAHsACgAJAAkAIgBpAGcAbgBvAHIAZQBVAG4AawBuAG8AdwBuACIAOgAgAGYAYQBsAHMAZQAKAAkAfQAsAAoACQAiAGYAbwByAG0AYQB0AHQAZQByACIAOgAgAHsACgAJAAkAIgBlAG4AYQBiAGwAZQBkACIAOgAgAHQAcgB1AGUALAAKAAkACQAiAGkAbgBkAGUAbgB0AFMAdAB5AGwAZQAiADoAIAAiAHQAYQBiACIACgAJAH0ALAAKAAkAIgBsAGkAbgB0AGUAcgAiADoAIAB7AAoACQAJACIAZQBuAGEAYgBsAGUAZAAiADoAIAB0AHIAdQBlACwACgAJAAkAIgByAHUAbABlAHMAIgA6ACAAewAKAAkACQAJACIAcgBlAGMAbwBtAG0AZQBuAGQAZQBkACIAOgAgAHQAcgB1AGUALAAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACIAbgB1AHIAcwBlAHIAeQAiADoAIAB7AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACIAbgBvAEYAbABvAGEAdABpAG4AZwBQAHIAbwBtAGkAcwBlAHMAIgA6ACAAIgBlAHIAcgBvAHIAIgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgAH0ACgAJAAkAfQAKAAkAfQAsAAoACQAiAGoAYQB2AGEAcwBjAHIAaQBwAHQAIgA6ACAAewAKAAkACQAiAGYAbwByAG0AYQB0AHQAZQByACIAOgAgAHsACgAJAAkACQAiAHEAdQBvAHQAZQBTAHQAeQBsAGUAIgA6ACAAIgBkAG8AdQBiAGwAZQAiAAoACQAJAH0ACgAJAH0ALAAKAAkAIgBhAHMAcwBpAHMAdAAiADoAIAB7AAoACQAJACIAZQBuAGEAYgBsAGUAZAAiADoAIAB0AHIAdQBlACwACgAJAAkAIgBhAGMAdABpAG8AbgBzACIAOgAgAHsACgAJAAkACQAiAHMAbwB1AHIAYwBlACIAOgAgAHsACgAJAAkACQAJACIAbwByAGcAYQBuAGkAegBlAEkAbQBwAG8AcgB0AHMAIgA6ACAAIgBvAG4AIgAKAAkACQAJAH0ACgAJAAkAfQAKAAkAfQAsAAoAIAAgACIAcABsAHUAZwBpAG4AcwAiADoAIABbAAoAIAAgACAAIAAiAC4ALwBwAGwAdQBnAGkAbgAuAGcAcgBpAHQAIgAKACAAIABdAAoAfQAKAA%3D%3D&files.main.ts=aQBtAHAAbwByAHQAIAB7ACAAdABlAHMAdAAgAH0AIABmAHIAbwBtACAAJwAuAC8AbABpAGIALgBqAHMAJwAKAAoAYQBzAHkAbgBjACAAZgB1AG4AYwB0AGkAbwBuACAAbQBhAGkAbgAoACkAOgAgAFAAcgBvAG0AaQBzAGUAPAB2AG8AaQBkAD4AIAB7AAoAIAAgAHQAZQBzAHQAKAApAAoAfQAKAAoAbQBhAGkAbgAoACkA&files.lib.ts=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIAB0AGUAcwB0ACgAKQA6ACAAUAByAG8AbQBpAHMAZQA8AGIAbwBvAGwAZQBhAG4APgAgAHsACgAgACAAcgBlAHQAdQByAG4AIABQAHIAbwBtAGkAcwBlAC4AcgBlAHMAbwBsAHYAZQAoAHQAcgB1AGUAKQA7AAoAfQA%3D

### Expected result

Show an error on the floating promise

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
This example seems to be working https://biomejs.dev/playground/?lintRules=noFloatingPromises&files.main.ts=aQBtAHAAbwByAHQAIAB7ACAAdABlAHMAdAAgAH0AIABmAHIAbwBtACAAJwAuAC8AbABpAGIALgBqAHMAJwAKAAoAYQBzAHkAbgBjACAAZgB1AG4AYwB0AGkAbwBuACAAbQBhAGkAbgAoACkAOgAgAFAAcgBvAG0AaQBzAGUAPAB2AG8AaQBkAD4AIAB7AAoAIAAgAHQAZQBzAHQAKAApAAoAfQAKAAoAbQBhAGkAbgAoACkA&files.lib.ts=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIAB0AGUAcwB0ACgAKQA6ACAAUAByAG8AbQBpAHMAZQA8AGIAbwBvAGwAZQBhAG4APgAgAHsACgAgACAAcgBlAHQAdQByAG4AIABQAHIAbwBtAGkAcwBlAC4AcgBlAHMAbwBsAHYAZQAoAHQAcgB1AGUAKQA7AAoAfQA%3D

So presence of the configuration breaks the playgorund. Can you please provide a minimal repository @Zer0PointModule ?
